sprint_08

Gvozdkov opened the pull request:
Понял ошибку с констрантами и добавил их к Like Button. Спасибо за информацию по градиенту! Постараюсь правильно реализовать после прохождения sprint_09.

Commit:
Импортированны файлы с "поломанного" проекта ImageFeed с внесенными изменениями от рекомендаций ревьюера для sprint_08

Commit:
1. Исправлены рекомендации по констрейнтам
2. Добавлен градиент в ячейку от черного к белому.
    Градиент реализован в ImagesListViewController при помощи ленивой переменной gradient.

Commit:
Добавлены констрейнты для Like Button

Commit:
Исправлены "поплывшие" констрейнты у Like button

IBDesignable commented on ImageFeed/Main.storyboard:
```diff
@@ -39,7 +39,12 @@
-                                                    <rect key="frame" x="356" y="14" width="42" height="42"/>
+                                                    <rect key="frame" x="354" y="4" width="44" height="44"/>
+                                                    <constraints>
+                                                        <constraint firstAttribute="height" constant="44" id="0dN-M5-DIC"/>
+                                                        <constraint firstAttribute="width" constant="44" id="LKy-rw-MLV"/>
+                                                        <constraint firstAttribute="width" secondItem="rbF-QK-TIa" secondAttribute="height" multiplier="1:1" id="pdS-rH-jQ3"/>
```
обычно оставляют что-то одно:
1) ширина-высота 
2) ширина-соотношение сторон или высота/соотношение сторон. 

Второй вариант лучше с той стороны, если размеры элемента придется править - то значение поправится одно, по соотношению элемент подтянется. Если же выбирать первый - то исправлений будет два. 

Оба варианта оставлять не желательно: слишком много ограничений, всегда обходись достаточным количеством

IBDesignable commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,95 @@
+        formatter.timeStyle = .none
+        return formatter
+    }()
+    
+//    private lazy var gradient: CAGradientLayer = {
+//        let gradient = CAGradientLayer()
+//        gradient.type = .axial
+//        gradient.colors = [
+//            UIColor.black.cgColor,
+//            UIColor.white.cgColor,
+//        ]
+//        gradient.startPoint = CGPoint(x: 0.0, y: 0.5)
+//        gradient.endPoint = CGPoint(x: 1.0, y: 0.5)
+//        return gradient
+//    }()
+
```
закомментированный код – это мертвый код, который загрязняет файл и мешает его восприятию, по мере накопления другого кода.

Нужно удалять перед сдачей на ревью, чтобы ни один ревьювер тебе такое замечание больше не оставлял

IBDesignable reviewed (approved):
С версткой все в порядке, перед тем как мерджить пул-реквест рекомендую почистить код и поправить его по комментариям к кодстайлу оставленным в предыдщих итерациях ревью

После этого – зеленый свет на мердж)

Commit:
Убрано перед Merge pull request:
  1. Лишний констрейн у кнопки Like Button  <constraint firstAttribute="width" constant="44" id="LKy-rw-MLV"/>
  2. Весь закомментированный код
